libs/sysplugins/smartycompilerexception.php

```diff
@@ -14,14 +14,14 @@ class SmartyCompilerException extends SmartyException
      * @param int            $code     The Exception code.
      * @param string|null    $filename The filename where the exception is thrown.
      * @param int|null       $line     The line number where the exception is thrown.
-     * @param Throwable|null $previous The previous exception used for the exception chaining.
+     * @param \Throwable|null $previous The previous exception used for the exception chaining.
      */
     public function __construct(
         string $message = "",
         int $code = 0,
         ?string $filename = null,
         ?int $line = null,
-        Throwable $previous = null
+        ?\Throwable $previous = null
```

Comment:
why did you add the backslash?

Reply:
Mostly because it was also added in the PR for V5 https://github.com/smarty-php/smarty/pull/1043/files#diff-d0f40f666506e3868faaf237c1f5a7731be2cd3c96bfa986b564f7c765b42bb5R26

But I can remove it

Reply:
Ah but I'm dumb there's no namespace in smarty 4, so not needed